kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/resolver/ProxyResolutionResult.java

```diff
@@ -32,26 +24,10 @@
  * describes which dependencies could not be resolved per VirtualKafkaCluster.
```

Comment:
The comment begins: 

> The result of a deep resolution of the dependencies of a single KafkaProxy.

I think this might be more consistent with the terminology we're using elsewhere: 

> The result of a deep resolution of the transitive references from a single KafkaProxy.

I note that you're using both "resolution" and "dereference". We should try to stick to one.

Reply:
purged all `deref` mentions and removed most usages of `dependency/dependencies`. I'll put up a subsequent PR to rename `DependencyResolver` to try and get away from the `dependency` noun altogether.